### PR TITLE
Jedwards/testsupdates

### PR DESCRIPTION
**Pull request by jedwards4b**
Begin to move tests from csh to python.   Beginning with test SMS.

### CODE REVIEW

File: scripts/Testing/Testcases/ICP_build.csh [hunk 1]
```diff
@@ -1,194 +0,0 @@
-#!/bin/csh -f
```
Comment (jedwards4b) [inline scripts/Testing/Testcases/ICP_build.csh]: Test builds all happen in python now.

File: scripts/Tools/bless_test_results [hunk 2]
```diff
@@ -261,7 +261,7 @@ def bless_test_results(baseline_name, test_root, compiler, test_id=None, namelis
                         broken_blesses.append((test_name, reason))
 
     # Make sure user knows that some tests were not blessed
-    for broken_bless in broken_blesses:
+    for broken_bless, reason in broken_blesses:
```
Comment (jedwards4b) [inline scripts/Tools/bless_test_results]: Not sure how this error was caught before now.

File: utils/python/CIME/XML/compilers.py [hunk 3]
```diff
@@ -4,18 +4,21 @@
...
 class Compilers(GenericXML):
 
-    def __init__(self, compiler=None, machine=None, os_= None, mpilib=None, infile=None):
+    def __init__(self, compiler=None, machine=None, os_= None, mpilib=None, infile=None, files=None):
         """
         initialize an object
         """
-        if (infile is None):
-            infile = os.path.join(get_cime_root(), "cime_config", get_model(), "machines", "config_compilers.xml")
+        if infile is None:
```
Comment (jedwards4b) [inline utils/python/CIME/XML/compilers.py]: Get the file name from config_files.xml, append local file if it exists

File: utils/python/CIME/build.py [hunk 4]
```diff
@@ -133,8 +133,11 @@ def post_build(case, logs):
         if not os.path.exists(bldlogdir):
             os.makedirs(bldlogdir)
 
-        for log in logs:
-            run_cmd("gzip %s" % log, from_dir=bldlogdir)
+    for log in logs:
```
Comment (jedwards4b) [inline utils/python/CIME/build.py]: The log files were not being copied back to bldlogdir, this corrects that.

File: utils/python/CIME/setup_tools.py [hunk 5]
```diff
@@ -1,182 +0,0 @@
-"""
```
Comment (jedwards4b) [inline utils/python/CIME/setup_tools.py]: moved to compilers.py

File: scripts/Tools/bless_test_results (hunk 2 above)
Comment (jgfouca) [inline scripts/Tools/bless_test_results]: Me either. Thanks for catching though.

**Comment by jgfouca** [comment]
Looks good.

**Pull request merged**